From 4d2e14e07282f2a121eb8108b195d43c11fcc105 Mon Sep 17 00:00:00 2001
From: Branden Archer <brarcher@google.com>
Date: Thu, 8 Apr 2021 23:11:27 -0700
Subject: [PATCH 3/3] Dump synthetic password related keys

The change relates to NIAP certification testing. This
should not be committed, as it should only be used in
one-off builds for testing.

Bug:133426426

Test: Created a lock screen password, observed keys dumped to logs
      from keystore and SyntheticPasswordManager
Change-Id: I80e369de887ce0ab3b79cc49d90d2f9f43b3504c
---
 .../server/locksettings/SyntheticPasswordManager.java     | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/services/core/java/com/android/server/locksettings/SyntheticPasswordManager.java b/services/core/java/com/android/server/locksettings/SyntheticPasswordManager.java
index 00ed8c6569ba..4b7acf159174 100644
--- a/services/core/java/com/android/server/locksettings/SyntheticPasswordManager.java
+++ b/services/core/java/com/android/server/locksettings/SyntheticPasswordManager.java
@@ -752,6 +752,10 @@ public class SyntheticPasswordManager {
         long handle = generateHandle();
         PasswordData pwd = PasswordData.create(credential.getType());
         byte[] pwdToken = computePasswordToken(credential, pwd);
+
+        String hexPwdToken = String.valueOf(HexEncoding.encode(pwdToken));
+        Log.i(TAG, "CKM.4.1 pwdToken " + hexPwdToken);
+
         final long sid;
         final byte[] applicationId;
 
@@ -1138,6 +1142,8 @@ public class SyntheticPasswordManager {
 
     private AuthenticationToken unwrapSyntheticPasswordBlob(long handle, byte type,
             byte[] applicationId, long sid, int userId) {
+        String hexApplicationId = String.valueOf(HexEncoding.encode(applicationId));
+        Log.i(TAG, "CKM.4.2 applicationId " + hexApplicationId);
         byte[] blob = loadState(SP_BLOB_NAME, handle, userId);
         if (blob == null) {
             return null;
@@ -1171,6 +1177,8 @@ public class SyntheticPasswordManager {
             }
             result.recreateFromEscrow(secret);
         } else {
+            String hexSyntheticPassword = String.valueOf(HexEncoding.encode(secret));
+            Log.i(TAG, "CKM.4.3 synthetic password " + hexSyntheticPassword);
             result.recreateDirectly(secret);
         }
         if (version == SYNTHETIC_PASSWORD_VERSION_V1) {
-- 
2.31.1.295.g9ea45b61b8-goog
